src/pages/index.js

```diff
@@ -64,50 +129,107 @@ function handleImageClick(link, name) {
   popupWithImage.openModal(link, name);
 }
 
-// SECTION - GALLERY LIST -----
-
-const galleryListSection = new Section(
-  {
-    items: initialCards,
-    renderer: (data) => {
-      const cardElement = createCard(data);
-
-      galleryListSection.addItem(cardElement);
-    },
-  },
-  ".gallery__list"
-);
-
-galleryListSection.renderItems();
-
 // ADD NEW CARD FORM -----
 
 const newCardFormPopup = new PopupWithForm({
   popupSelector: "#add-modal",
   handleFormSubmit: handleAddCardFormSubmit,
 });
 
-profileAddButton.addEventListener("click", () => newCardFormPopup.openModal());
+function handleDeleteClick(cardData) {
+  confirmPopup.openModal(cardData);
+}
+
+profileAddButton.addEventListener("click", () => {
+  cardFormValidator.resetValidation();
+  newCardFormPopup.openModal();
+});
 
 function createCard(data) {
-  const card = new Card(data, "#card-template", handleImageClick);
+  const card = new Card(
+    data,
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
   const cardElement = card.generateCard();
   return cardElement;
 }
 
+function handleLikeClick(card) {
+  const cardId = card.getId();
+  if (card.getLikeStatus()) {
+    api
+      .unlikeCard(cardId)
+      .then(() => {
+        card.setLikeStatus(false);
+      })
+      .catch((err) => console.error(`${err} Something went wrong`));
+  } else {
+    api
+      .likeCard(cardId)
+      .then(() => {
+        card.setLikeStatus(true);
+      })
+      .catch((err) => console.error(`${err} Something went wrong`));
+  }
+}
+
 newCardFormPopup.setEventListeners();
 
 function handleAddCardFormSubmit(inputValues) {
-  const cardElement = createCard(inputValues);
-  galleryListSection.addItem(cardElement);
-  newCardFormPopup.closeModal();
-  cardFormValidator.toggleButtonState();
+  newCardFormPopup.setButtonText("Saving...");
+  api
+    .addNewCard(inputValues.name, inputValues.link)
+    .then((data) => {
+      const cardElement = createCard(data);
+      galleryListSection.addItem(cardElement);
+      newCardFormPopup.closeModal();
+      cardFormValidator.toggleButtonState();
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      newCardFormPopup.setButtonText("Create");
+    });
+}
+
+//CONFIRMATION POPUP -----
+
+function handleSubmit(card) {
+  const cardId = card.getId();
+  api
+    .deleteCard(cardId)
+    .then(() => {
+      card.removeCard();
+    })
+    .catch((err) => console.error(err));
+  confirmPopup.closeModal();
 }
 
+const confirmPopup = new ConfirmPopup({
+  popupSelector: "#confirmationModal",
+  handleSubmit,
+});
+
+confirmPopup.setEventListeners();
+
 //FORM INIT -----
 
 const editFormValidator = new FormValidator(options, editProfileForm);
 const cardFormValidator = new FormValidator(options, addCardForm);
+const avatarFormValidator = new FormValidator(options, avatarForm);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form
    const formName = formElement.getAttribute('name')

   // here you store the validator using the `name` of the form
    formValidators[formName] = validator;
   validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```